CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+Welcome to Espanso!
+We are very happy to have you and we thank you for considering contributing!
+
+We would like to order the contributions like this:
+
+- Before you start hacking, it is important to **ask first** if your idea or bugfix is in order.
+  Create an issue or come and say hi in the `#dev` channel for fixing bugs and adding features involved with coding logic, and `#documentation` for updating primarly the website. Join us at [the discord][`espanso` discord].
+  We don't bite!.
+
+  It would be sad that you do the effort to clone the project, successfully make the PR, but it wasn't in our plans or there is another PR that is currently adressing that issue.
+
+- After the PR is submitted, the workflows will start to lint, check and test the changes. Please try to stay all green ✅.
+- Most of the time we take some time to respond. Sorry, we are few and there SO much to do here!
+
+## General guidelines and philosophy
+
+This is a list of things we would like to have and mantain across time. Please do your best to abide by.
+
+- We are geared towards a full-rust codebase, but today there is quite a bit of C++ code. We can't get rid of it today, but over time we will prefer to stay in the rust side.
+- Everything should be explained on the documentation via drawings, markdown files, etc, but it is important to make it clear. There will always be some new guy or gal into the project we want to welcome 😄.
+- Use clear variable names and try to avoid confusing abbreviations. Think that your peers may not be fully fluent in english 💬.
+
+[`espanso` discord]: https://discord.gg/4QARseMS6k
+
+## Developing
+
+We would like the rust code:
+
+- to be formatted via `rustfmt`
+  ```console
+  cargo fmt --all
+  ```
+- to abide by the `clippy` ruleset we currently use
+  ```console
+  cargo clippy --workspace
+  ```
+- to approve the tests by running
+  ```console
+  cargo test --workspace
+  ```
+- to be compiled with `stable`, not `nightly`
+- prefer not to use macros, if possible. Try to use functions or generics.
+
+And C / C++ code:
+
+- _which formatting?_
+- _do we have lints?_
+- _any suggestions?_
```

Comment:
That's definitely an area we could improve :)

Reply:
I suggested `clang-tidy` and `clang-format`. There is a problem that `clang-*` doesn't work with a folder recusively, you need to pass file by file. I lookup for a rust wrapper (so we don't work with bash scripts) and found one!
[run-clang-tidy](https://crates.io/crates/run-clang-tidy) and [rust-clang-format](https://github.com/lmapii/run-clang-format)
What do you think?

Reply:
Looks promising :)